src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -234,6 +251,63 @@ public static Text of(Object... objects) {
             }
         }
 
+        if (builder.children.size() == 1) {
+            // Single content, reduce Text depth
+            return builder.children.get(0);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Builds a {@link Text} from a given array of content objects. This method
+     * assumes, that all elements of the array are content and not formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param objects The object array
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Object... objects) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : objects) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
```

Comment:
I will push a "fix" / a change of behaviour for `Texts.of(Object...)` soon. 
(I waited for gratimax, to discuss that, but he is currently afk, 
so will i propose it and I'll try to talk again with him about it later)

However, this method is not meant to join recursively. 
I'm not sure whether this is necessary at all, do you have an example usecase?

If there is a valid usecase the method will most likely look like this:
````java
public static Text deepJoinOf(Text separator, Object... objects) {
	TextBuilder builder = builder();
	boolean first = true;

	for (Object part : objects) {
	   if (!first) {
			builder.append(separator);
		}
		first = false;
		if (part instanceOf Object[]) {
			builder.append(Texts.deepJoinOf(separator, (Object[]) part));
		} else if (part instanceOf Iterable) {
			builder.append(Texts.deepJoinOf(separator, (Iterable<?>) part));
		} else {
			builder.append(Texts.of(part));
		}
	}
	return builder.build();
}
````